convert2rhel/checks.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright(C) 2016 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+import logging
+
+from convert2rhel.pkghandler import call_yum_cmd
+
+logger = logging.getLogger(__name__)
+
+
+def check_repos_are_valid():
+    output, ret_code = call_yum_cmd(command="repolist", args="-v")
+    if "failure" in output or ret_code != 0:
+        logger.critical("Couldn't validate custom repositories. Please check custom repo config files.")
+    logger.debug("All custom repo appears to be valid.")
```

Comment:
We need to validate that the repos the user has passed with `--enablerepo` is a valid repo, not that all the repos are valid.

Adding to that, this is the output I get with misconfigured URL when running `repolist`, the check wouldn't catch that
```
[root@c2r-20210305121855 ~]# yum repolist -v
Failed to set locale, defaulting to C
Loading "product-id" plugin
Loading "search-disabled-repos" plugin
Loading "subscription-manager" plugin
Updating Subscription Management repositories.
Config time: 3.201
Yum Version: 3.2.29
Repository rhel-6-server-rpms is listed more than once in the configuration
Repository rhel-6-server-optional-rpms is listed more than once in the configuration
Repository rhel-6-server-extras-rpms is listed more than once in the configuration
https://download.caopr.fedorainfracloud.org/results/%40oamg/convert2rhel/epel-6-x86_64/repodata/repomd.xml: [Errno 14] PYCURL ERROR 6 - "Couldn't resolve host 'badurl.copr.fedorainfracloud.org'"
Trying other mirror.
Setting up Package Sacks
pkgsack time: 0.040
Repo-id      : copr:copr.fedorainfracloud.org:group_oamg:convert2rhel
Repo-name    : Copr repo for convert2rhel owned by @oamg
Repo-revision: 1614936293
Repo-updated : Fri Mar  5 10:24:53 2021
Repo-pkgs    : 54
Repo-size    : 5.0 M
Repo-baseurl : https://badurl.copr.fedorainfracloud.org/results/@oamg/convert2rhel/epel-6-x86_64/
Repo-expire  : 21600 second(s) (last: Fri Mar  5 11:21:47 2021)
Repo-excluded: 54

Repo-id      : rhel-6-server-rpms
Repo-name    : Red Hat Enterprise Linux 6 Server (RPMs)
Repo-revision: 1609129994
Repo-updated : Mon Dec 28 05:33:13 2020
Repo-pkgs    : 21210
Repo-size    : 48 G
Repo-baseurl : https://cdn.redhat.com/content/dist/rhel/server/6/6Server/x86_64/os
Repo-expire  : 86400 second(s) (last: Fri Mar  5 11:26:57 2021)

repolist: 21264

[root@c2r-20210305121855 ~]# echo $?
0

```

Reply:
Hi @j-mracek Did we have some way to check a specific repo using yum ?

Reply:
Please could you provide additional information. Did you mean yum in RHEL7 or RHEL8? It is you but two different component.
What do you suggest as a check? Repo is well defined, available on server, data are downloadable or something else?

Reply:
@j-mracek
I need to validate if individual repo are valid, in other word: if I can use this repo for download and install packages, for example: we had 6 repos listed on config file, and I need to test if one of that are valid, and I can use to download/install packages.

Reply:
Using `makecache --setopt=reposdir=/etc/yum.repos.d -v --setopt=*.skip_if_unavailable=False` as @j-mracek suggestion